Svc/GenericHub/GenericHubComponentImpl.cpp

```diff
@@ -96,13 +94,85 @@ void GenericHubComponentImpl ::dataIn_handler(const NATIVE_INT_TYPE portNum,
         status = wrapper.setBuffLen(rawSize);
         FW_ASSERT(status == Fw::FW_SERIALIZE_OK, static_cast<NATIVE_INT_TYPE>(status));
         portOut_out(port, wrapper);
+        // Deallocate the existing buffer
         dataInDeallocate_out(0, fwBuffer);
     } else if (type == HUB_TYPE_BUFFER) {
+        // Fw::Buffers can reuse the existing data buffer as the storage type!  No deallocation done.
         fwBuffer.set(rawData, rawSize, fwBuffer.getContext());
         buffersOut_out(port, fwBuffer);
+    } else if (type == HUB_TYPE_EVENT) {
+        FwEventIdType id;
+        Fw::Time timeTag;
+        Fw::LogSeverity severity;
+        Fw::LogBuffer args;
+
+        // Deserialize tokens for events
+        status = incoming.deserialize(id);
+        FW_ASSERT(status == Fw::FW_SERIALIZE_OK, static_cast<NATIVE_INT_TYPE>(status));
```

Comment:
Are you assuming that the upstream device reliably decodes the data, and that it will never pass you a partial packet? If no, then you should assert, as that makes the software assert on a potentially unreliable communication channel.

Reply:
Yes, I am making that assumption.  GenericHub is designed to sit on top of framer and deframer and those two pieces ensure proper packet transmission.